Godeps/_workspace/src/k8s.io/kubernetes/pkg/controller/framework/shared_informer.go

```diff
@@ -203,16 +207,35 @@ func (s *sharedIndexInformer) AddEventHandler(handler ResourceEventHandler) erro
 	s.startedLock.Lock()
 	defer s.startedLock.Unlock()
 
-	if s.started {
-		return fmt.Errorf("informer has already started")
+	if !s.started {
+		listener := newProcessListener(handler)
+		s.processor.listeners = append(s.processor.listeners, listener)
+		return nil
 	}
 
+	// in order to safely join, we have to
```

Comment:
@smarterclayton @liggitt 

Let's get some comments on whether this works or not.  I'm pretty sure that this will do what I expect, since the indexer can't be updated until the block on deltas is lifted.  This is likely to be costly, but its not going to be common and it helps unwind the: 

 1. initializing controllers requires that ensureSA is completed
 2.  which requires quota admission
 3.  which requires quota and namespace caches to be filled
 4.  which requires quota and namespace informers to have been started
 5. which requires all the controllers to be initialized

problem I was hitting.